tests/test_env.py

```diff
@@ -894,3 +895,46 @@ def test_open_file_expired_aws_credentials(monkeypatch, caplog, path_rgb_byte_ti
         with rasterio.env.Env():
             with rasterio.open(path_rgb_byte_tif) as dataset:
                 assert not dataset.closed
+
+
+@pytest.fixture
+def http_server(tmpdir):
+    import functools
+    import multiprocessing
+    import http.server
+    from . import rangehttpserver
+    PORT = 8000
+    Handler = functools.partial(rangehttpserver.RangeRequestHandler, directory=str(tmpdir))
+    httpd = http.server.HTTPServer(("", PORT), Handler)
+    p = multiprocessing.Process(target=httpd.serve_forever)
```

Comment:
All my attempts at launching a server in a child thread failed and I resorted to a sub-process.